src/main/java/forum/rest/ThreadRestController.java

```diff
@@ -0,0 +1,39 @@
+package forum.rest;
+
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import forum.entity.Category;
+import forum.entity.Thread;
+import forum.service.ThreadService;
+
+@RestController
+@RequestMapping("/api")
+public class ThreadRestController {
+
+	@Autowired
+	private ThreadService threadService;
+
+	@Autowired
+	private CategoryRestController categoryController;
+
+	@PostMapping("/threads")
+	public Thread addThread(@RequestBody Thread theThread) {
+
+		theThread.setId(0);
```

Comment:
What does this line mean? What is the purpose?

Reply:
This is done just in case an id is passed in JSON. It will set id to 0 so it will save a new item instead update an existing one.

Reply:
In the course the instructor did so, but I'm not sure if it's useful here since it's for POST method, not PUT. Maybe it's so that we can use the same methods in Service and DAO for the PUT method in the future? Or does it matter even for that?